x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/AnalyzerTests.java

```diff
@@ -2603,6 +2610,137 @@ public void testFunctionNamedParamsAsFunctionArgument() {
         assertEquals(DataType.DOUBLE, ee.dataType());
     }
 
+    public void testResolveInsist_fieldExists_insistedOutputContainsNoUnmappedFields() {
+        assumeTrue("Requires UNMAPPED FIELDS", EsqlCapabilities.Cap.UNMAPPED_FIELDS.isEnabled());
+
+        LogicalPlan plan = analyze("FROM test | INSIST_🐔 emp_no");
+
+        assertThat(plan.output().getLast().name(), is("emp_no"));
+        assertThat(
+            plan.output()
+                .stream()
+                .filter(a -> a instanceof FieldAttribute fa && fa.field() instanceof PotentiallyUnmappedKeywordEsField)
```

Comment:
A stronger check here is that the original `emp_no` with its type (integer) from `test` is kept as is.